### ISSUE TITLE
Loosen constraint on debase-ruby_core_source version

### ISSUE THREAD

**Issue by mscrivo** [DataDog/dd-trace-rb#4058]
We're trying to upgrade to [debase 0.26](https://github.com/ruby-debug/debase/releases/tag/0.2.6), which finally fixes the compilation issues on recent versions macOS/xcode, but we've run into this issue:

```
Could not find compatible versions

Because debase >= 0.2.6, < 3.0.0.beta.12 depends on debase-ruby_core_source >= 3.3.6
  and every version of datadog depends on debase-ruby_core_source = 3.3.1,
  debase >= 0.2.6, < 3.0.0.beta.12 is incompatible with datadog >= 0.
So, because Gemfile depends on datadog >= 0
  and Gemfile depends on debase ~> 0.2.6,
  version solving has failed.
```

Would it be possible to loosen the constraint on `debase-ruby_core_source`?

**Comment by ivoanjo** [comment]
Hey @mscrivo thanks for reaching out as usual and sorry for the sharp edge 👀 

TL;DR for the next dd-trace-rb version we're actually replacing the `debase-ruby_core_source` dependency with our own fork (see https://github.com/DataDog/dd-trace-rb/pull/4014 ). So you should be good to go with 2.5.0 once it's out.

(Non TL;DR version is: We wanted to trim some of the unused headers + be able to do a few tweaks that we don't think make sense to submit upstream as `debase` has different needs)

**Comment by mscrivo** [comment]
Thanks!
